src/runtime/containerd-shim-v2/service.go

```diff
@@ -917,6 +917,11 @@ func (s *service) Shutdown(ctx context.Context, r *taskAPI.ShutdownRequest) (_ *
 
 	s.cancel()
 
+	// Since we only send an shutdown qmp command to qemu when do stopSandbox, and
+	// didn't wait until qemu process's exit, thus we'd better to make sure it had
+	// exited when shimv2 terminated. Thus here to do the last cleanup of the hypervisor.
+	syscall.Kill(int(s.hpid), syscall.SIGKILL)
```

Comment:
- This is technically redundant if agent tracing is enabled since in that scenario we wait for the process to end (see #1697), however I don't think this change is going to cause a problem.
- I wonder if we should log the return code from `Kill()` if debug is enabled?

Reply:
OK, I see this has been merged. But I'm a bit concerned about the root cause for qemu not responding to the shutdown command.

Reply:
Hi @c3d 

Just as I commented above, the qemu actually had responsed to the shutdown command and had sent out an event as event="{SHUTDOWN map[guest:false reason:host-qmp-quit] 2021-07-08 04:17:58.000200121 +0000 UTC}" , but it do need a while to gracefully exit, but we didn't give it the time to exit. That's why the CI would see a "temporary leak" of qemu process after the shimv2 had exited.